### ISSUE TITLE
SC2006 triggered with quotes and backtics misinterpreted

### ISSUE THREAD

**Issue by onkobu** [koalaman/shellcheck#3009]
#### For bugs
- Rule Id SC2006

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
local count="$(aws s3api list-buckets --output text --query 'Buckets[?starts_with(Name,`'"$bucketPattern"'`)].Name' | wc -w)"

```

#### Here's what shellcheck currently says:

SC2006 (style): Use $(...) notation instead of legacy backticks `...`

#### Here's what I wanted or expected to see:

No warning, The backticks are in two separate strings with even a variable in between. AWS CLI requires this syntax with backticks in --query parameter. Basically this combines a double quoted string containing a single quote and a backtick with a variable and a second string closing backticks and single quote.
